conda_smithy/lint_recipe.py

```diff
@@ -64,7 +64,13 @@ def lintify(meta, recipe_dir=None):
         if bad_selectors:
             lints.append('Selectors are suggested to take a "  # [<selector>]" '
                          'form.')
-    
+
+    # 7: The build section should have a build number.
+    build_section = meta.get('build', {}) or {}
+    build_number = build_section.get('number', None)
+    if build_number is None:
+        lints.append('The recipe must have a `build/number` section.')
+
```

Comment:
Not sure if I got what you maintained in https://github.com/conda-forge/conda-smithy/pull/127#discussion_r59376022. Is this what you had in mind?